.github/workflows/release.yml

```diff
@@ -0,0 +1,188 @@
+
+      - name: DEBUG
+        shell: bash
+        run: |
+          echo "Tag & Platform is '${{ env.TAG_NAME }}-${{ env.PLATFORM }}'"
+
+      - name: Build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release
+
+      - name: Build MacOS aarch64 target
+        if: matrix.operating-system == 'macos-11'
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --target aarch64-apple-darwin
+
+      - name: Compress
+        uses: thedoctor0/zip-release@0.6.2
+        with:
+          type: 'zip'
+          directory: 'target/release/'
+          path: 'creditcoin-node*'
+          filename: '../../creditcoin-${{ env.TAG_NAME }}-${{ env.PLATFORM }}.zip'
+          exclusions: 'creditcoin-node.d'
+
+      - name: Compress MacOS aarch64 target
+        if: matrix.operating-system == 'macos-11'
+        uses: thedoctor0/zip-release@0.6.2
+        with:
+          type: 'zip'
+          directory: 'target/aarch64-apple-darwin/release/'
+          path: 'creditcoin-node*'
+          filename: '../../../creditcoin-${{ env.TAG_NAME }}-aarch64-apple-darwin.zip'
+          exclusions: 'creditcoin-node.d'
+
+      - name: Upload binary
+        uses: actions/upload-artifact@v3
+        with:
+          path: 'creditcoin-${{ env.TAG_NAME }}-*.zip'
+          if-no-files-found: error
+
+  build-wasm-runtime:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Figure out tag name
+        shell: bash
+        run: |
+          TAG_NAME=$(git describe --tag)
+          echo "TAG_NAME=$TAG_NAME" >> $GITHUB_ENV
+
+      - name: Build WASM runtime
+        id: srtool_build
+        uses: chevdor/srtool-actions@v0.5.0
+        with:
+          chain: "creditcoin"
+          runtime_dir: runtime
+          package: creditcoin-node-runtime
+          workdir: ${{ github.workspace }}
+
+      - name: DEBUG
+        run: |
+          echo '${{ steps.srtool_build.outputs.json }}' | jq . > creditcoin-srtool-digest.json
+          cat creditcoin-srtool-digest.json
+          echo "Runtime location: ${{ steps.srtool_build.outputs.wasm }}"
+
+      - name: Rename file
+        run: |
+          mv ${{ steps.srtool_build.outputs.wasm }} creditcoin-${{ env.TAG_NAME }}-runtime.wasm
+
+      - name: Upload binary
+        uses: actions/upload-artifact@v3
+        with:
+          path: '*.wasm'
+          if-no-files-found: error
+
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Figure out tag name
+        shell: bash
+        run: |
+          TAG_NAME=$(git describe --tag)
+          echo "TAG_NAME=$TAG_NAME" >> $GITHUB_ENV
+
+      - name: Build docker image
+        run: |
+          docker build -t gluwa/creditcoin-substrate:${{ env.TAG_NAME }} .
+    # TODO: docker push
```

Comment:
@dlebee I see there's some logic to publishing tags in the Azure pipeline. Can you explain what is it?